### PR TITLE
[graphviz] update to 9.0.0

### LINKED ISSUE #33199: [graphviz] update to 8.1.0
### Library name

graphviz

### New version number

8.1.0

### Other information that may be useful (release notes, etc...)

Release: https://gitlab.com/graphviz/graphviz/-/releases/8.1.0  
Changelog: https://gitlab.com/graphviz/graphviz/raw/main/CHANGELOG.md

### PR DESCRIPTION
**Pull request by talregev**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by talregev** [comment]
There is a problem with post process (found exe on bin and debug/bin) Not sure how to handle. I will apricate the help. 
```cmd
warning: The following EXEs were found in /bin or /debug/bin. EXEs are not valid distribution targets.

    C:\src\vcpkg\packages\graphviz_x64-windows\bin\cluster.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\bin\dot_builtins.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\bin\gv2gxl.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\bin\gxl2dot.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\bin\prune.exe

warning: The following EXEs were found in /bin or /debug/bin. EXEs are not valid distribution targets.

    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\cluster.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\dot_builtins.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\gv2gxl.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\gxl2dot.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\prune.exe

error: Found 2 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: C:\src\vcpkg\ports\graphviz\portfile.cmake
```

**Comment by talregev** [comment]
On the CI it fail with:
```
CMake Error at D:/downloads/tools/cmake-3.27.1-windows/cmake-3.27.1-windows-i386/share/cmake-3.27/Modules/FindPkgConfig.cmake:607 (message):
  A required package was not found
Call Stack (most recent call first):
  D:/downloads/tools/cmake-3.27.1-windows/cmake-3.27.1-windows-i386/share/cmake-3.27/Modules/FindPkgConfig.cmake:829 (_pkg_check_modules_internal)
  CMakeLists.txt:128 (pkg_check_modules)
```
I am not sure why. Any help?

**Comment by dg0yt** [comment]
> CMakeLists.txt:128 (pkg_check_modules)

Check that line to find the missing dependency.

File: ports/graphviz/vcpkg.json [hunk 1]
```diff
@@ -23,11 +22,13 @@
         "tiff"
       ]
     },
+    "librsvg",
     "pango",
     {
       "name": "pkgconf",
       "host": true
     },
+    "qt5",
```
Comment (dg0yt) [inline ports/graphviz/vcpkg.json:31]: No, never, not ALL of Qt5.

**Comment by talregev** [comment]
> > CMakeLists.txt:128 (pkg_check_modules)
> 
> Check that line to find the missing dependency.

```cmake
127: find_package(PkgConfig)
128: if(PkgConfig_FOUND)
```

It search for pkgconfig, and it don't find it. Something changed?

**Comment by dg0yt** [comment]
I cited the `pkg_check_modules` to narrow the search. Mind the patches. Probably this change in your PR:
~~~
pkg_check_modules(librsvg REQUIRED IMPORTED_TARGET librsvg)

**Comment by talregev** [comment]
> I cited the `pkg_check_modules` to narrow the search. Mind the patches. Probably this change in your PR:
> 
> ```
> pkg_check_modules(librsvg REQUIRED IMPORTED_TARGET librsvg)
> ```

It actually was the:
```
128:  pkg_check_modules(GTS gts REQUIRED)
```

Thank you for your help!

**Comment by talregev** [comment]
Now I have again the problem:

There is a problem with post process (found exe on bin and debug/bin) Not sure how to handle. I will apricate the help. 
```cmd
warning: The following EXEs were found in /bin or /debug/bin. EXEs are not valid distribution targets.

    C:\src\vcpkg\packages\graphviz_x64-windows\bin\cluster.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\bin\dot_builtins.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\bin\gv2gxl.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\bin\gxl2dot.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\bin\prune.exe

warning: The following EXEs were found in /bin or /debug/bin. EXEs are not valid distribution targets.

    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\cluster.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\dot_builtins.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\gv2gxl.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\gxl2dot.exe
    C:\src\vcpkg\packages\graphviz_x64-windows\debug\bin\prune.exe

error: Found 2 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: C:\src\vcpkg\ports\graphviz\portfile.cmake
```

**Comment by dg0yt** [comment]
If there are new tools, you have to update the `vcpkg_copy_tools` line.

**Comment by talregev** [comment]
> If there are new tools, you have to update the `vcpkg_copy_tools` line.

Thank you, that help!!

File: ports/graphviz/vcpkg.json (hunk 1 above)
Comment (talregev) [reply ports/graphviz/vcpkg.json]: removed.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All feature are tested successfully in the following triplet:
- x86-windows
- x64-windows

File: ports/graphviz/portfile.cmake [hunk 2]
```diff
@@ -84,7 +86,41 @@ foreach(script_or_link IN ITEMS "dot2gxl${VCPKG_TARGET_EXECUTABLE_SUFFIX}" gvmap
...
+        sccmap
+        sfdp
+        tred
+        twopi
+        unflatten
+        cluster
+        dot_builtins
+        gv2gxl
+        gxl2dot
+        prune
```
Comment (dg0yt) [inline ports/graphviz/portfile.cmake:123]: It used to be alphabetic.

File: ports/graphviz/cpp-error.patch [hunk 3]
```diff
@@ -0,0 +1,12 @@
+diff --git a/cmd/gvedit/mainwindow.cpp b/cmd/gvedit/mainwindow.cpp
+--- a/cmd/gvedit/mainwindow.cpp
++++ b/cmd/gvedit/mainwindow.cpp
```
Comment (dg0yt) [inline ports/graphviz/cpp-error.patch:3]: But gvedit is exlicitly disabled...

**Pull request merged**